### PR TITLE
Add INT4 support to DepthwiseConv and FullyConnected

### PR DESCRIPTION
**Pull request by paulinesho**
BUG=b/250680652

This PR adds INT4 filter support and tests for DepthwiseConv and FullyConnected operators.

### CODE REVIEW

File: tensorflow/lite/micro/kernels/fully_connected_test.cc [hunk 1]
```diff
@@ -42,6 +42,17 @@ const float simple_weights_data[] = {
     1, 2, 3, 4, 5, 6, 7, 8, 9, 10,  // u = 1
     1, 2, 3, 4, 5, 6, 7, 8, 9, 10,  // u = 2
 };
+// INT4 isn't supported on Hexagon (b/258710417)
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/fully_connected_test.cc:45]: let's make this `TODO(b/258710417): INT4 isn't currently supported on Hexagon.` That way the bug will have a link to all the TODO references in the code.

File: tensorflow/lite/micro/kernels/fully_connected_test.cc [hunk 2]
```diff
@@ -620,4 +632,38 @@ TF_LITE_MICRO_TEST(SimpleTestQuantizedInt8NullBias) {
       kTfLiteOk);
 }
 
+// INT4 isn't supported on Hexagon (b/258710417)
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/fully_connected_test.cc:635]: let's make this `TODO(b/258710417): INT4 isn't currently supported on Hexagon.` That way the bug will have a link to all the TODO references in the code.

**Review by advaitjain — COMMENTED** [review]
looks good, only small nits.

File: tensorflow/lite/micro/kernels/fully_connected_test.cc (hunk 1 above)
Comment (paulinesho) [reply tensorflow/lite/micro/kernels/fully_connected_test.cc]: Fixed.

File: tensorflow/lite/micro/kernels/fully_connected_test.cc (hunk 2 above)
Comment (paulinesho) [reply tensorflow/lite/micro/kernels/fully_connected_test.cc]: Fixed.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**